angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
A big problem with these solutions:
```ts
declare function f(): number[] | null;
const x = orderByFilter(f(), x => x ? x * 2 : 0);
```

`Argument of type 'number[] | null' is not assignable to parameter of type 'number[]'.`